source/opt/instruction.h

```diff
@@ -91,6 +91,46 @@ inline bool operator!=(const Operand& o1, const Operand& o2) {
   return !(o1 == o2);
 }
 
+// This structure is used to represent a DebugScope instruction from
+// the OpenCL.100.DebugInfo extened instruction set. Since this
+// structure is added to Instruction class as a member variable, we
+// want to reduce the size of structure. Hence, we keep only its Scope
+// and InlinedAt operands but not its result id. Note that losing its
+// result id will result in a different binary even when there is no
+// optimization. We skip the sanity check in Optimizer::Run(..) when
+// the module contains a DebugScope instruction.
```

Comment:
You should explain that the result id is not suppose to be used for anything, which is why we can ignore it.

Do we have a validator check that the result id of a debug scope instruction is not used?